cpp/include/cudf/detail/iterator.cuh

```diff
@@ -313,6 +313,36 @@ struct scalar_pair_accessor : public scalar_value_accessor<Element> {
   }
 };
 
+/**
+ * @brief Pair accessor for scalar's representation value and validity.
+ *
+ * @tparam Element The type of element in the scalar.
+ */
+template <typename Element>
+struct scalar_representation_pair_accessor : public scalar_value_accessor<Element> {
+  using super_t    = scalar_value_accessor<Element>;
+  using rep_type   = typename Element::rep;
```

Comment:
This will fail when `Element::rep` doesn't exist. It would be nice if I could generically use `make_rep_iterator` on any fixed-width type and for it to "just work".